### PR TITLE
refactor selftests with `script` wrapper, clean build

### PR DESCRIPTION
**Pull request by dannasessha**
I believe this commit is suitable to demonstrate a `checkmate` selftest _failure_, in the `audit` stage.

It also creates a plaintext log file (about 2k) of the entire `checkmate` run's terminal output, which could be an inroad towards reporting, or could be easily trimmed out.

### CODE REVIEW

File: tests/checktests.rs [hunk 1]
```diff
@@ -1,29 +1,42 @@
-mod common;
-use std::str;
+#[cfg(test)]
+mod selftest {
+    mod common;
+    use std::str;
+    #[test]
+    #[cfg_attr(not(feature = "selftest"), ignore)]
+    fn attempt_generic_output() {
```
Comment (zancas) [inline tests/checktests.rs:7]: If this is a sanity check that was written to help set up tests, but it's not testing the functionality of `cargo-checkmate`....

File: tests/selftest/common.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        .status()
+        .expect("cargo clean failed");
+    Command::new("cargo")
+        .arg("build")
+        .arg("-q")
+        .status()
+        .expect("cargo build failed");
+}
+
+pub fn attempt_output() -> Output {
```
Comment (zancas) [inline tests/selftest/common.rs:18]: Shall we clip this as well?

File: tests/checktests.rs (hunk 1 above)
Comment (dannasessha) [reply tests/checktests.rs]: Correct, I'll trim it.

File: tests/selftest/common.rs (hunk 2 above)
Comment (dannasessha) [reply tests/selftest/common.rs]: Yes, sounds good.

**Comment by zancas** [comment]
This is looking pretty good.

I would like to start learning about the effect of the tests.

I propose that we add a unittest, and exactly which unittest to add is defined by the steps I describe here. 

I propose that @dannasessha does the following:

(1) run tarpaulin against master
(2) run tarpaulin against this version of the code
(3) check out a new branch that:
   i.   experiments with a version of `selftest` that does _NOT_ run the test phase
   ii.  if such a version of `selftest` can be made to run, then compare the `tarpaulin` results of the 2 versions (selftest-that-runs-the-test-phase versus, selftest-that-does-not)

The product of the above proposed experiment will be (at least) deeper insight into how the (different) tests are covering the codebase.

If a `selftest` can be constructed, that fails the `structopt-version` of the code, _without_ running the tests twice, then:
    * tarpaulin _may_ show us which code is "uncovered" by the new approach
    * if we can learn that, then we can begin to construct a unit-test that targets exactly that code

Such a unit-test, will be viable-and-valuable regardless of which `selftest` version we eventually use.

File: tests/selftests.rs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+mod common;
+#[cfg(test)]
```
Comment (zancas) [inline tests/selftests.rs:2]: Can you put the `cfg_attr` ignore fu on the mod, a single time, and then ignore the whole mod?

**Comment by zancas** [comment]
Superseded by #3

**Pull request closed without merging**